presto-main/src/main/java/com/facebook/presto/operator/PageProcessor.java

```diff
@@ -24,13 +23,9 @@
 
 public interface PageProcessor
 {
-    int process(ConnectorSession session, Page page, int start, int end, PageBuilder pageBuilder);
-
     /**
      * @return filtered and projected page, null if all positions are filtered
      */
     @Nullable
-    Page processColumnar(ConnectorSession session, Page page, List<? extends Type> types);
-
-    Page processColumnarDictionary(ConnectorSession session, Page page, List<? extends Type> types);
+    Page process(ConnectorSession session, Page page, List<? extends Type> types);
```

Comment:
Btw this interface with returning `null` is somewhat ugly.
Can't we just return a Page with positionCount = 0?